lib/ex_webrtc/rtp/jitter_buffer.ex

```diff
@@ -0,0 +1,166 @@
+  """
+
+  # Heavily inspired by:
+  # https://github.com/membraneframework/membrane_rtp_plugin/blob/23f3279540aea7dea3a194fd5a1680c2549aebae/lib/membrane/rtp/jitter_buffer.ex
+
+  alias ExWebRTC.RTP.JitterBuffer.PacketStore
+  alias ExRTP.Packet
+
+  @default_latency_ms 200
+
+  @typedoc """
+  Options that can be passed to `new/1`.
+
+  * `latency` - latency introduced by the buffer, in milliseconds. `#{@default_latency_ms}` by default.
+  """
+  @type options :: [latency: non_neg_integer()]
+
+  @typedoc """
+  Time (in milliseconds) after which `handle_timer/1` should be called.
+  Can be `nil`, in which case no timer needs to be set.
+  """
+  @type timer :: non_neg_integer() | nil
+
+  @typedoc """
+  The 3-element tuple returned by all functions other than `new/1`.
+
+  * `packets` - a list with packets flushed from the buffer as a result of the function call. May be empty.
+  * `timer_duration_ms` - see `t:timer/0`.
+  * `buffer` - `t:#{inspect(__MODULE__)}.t/0`.
+
+  Generally speaking, all results of this type can be handled in the same way.
+  """
+  @type result :: {packets :: [Packet.t()], timer_duration_ms :: timer(), buffer :: t()}
+
+  @opaque t :: %__MODULE__{
+            latency: non_neg_integer(),
+            store: PacketStore.t(),
+            state: :initial_wait | :timer_set | :timer_not_set
+          }
+
+  @enforce_keys [:latency]
+  defstruct @enforce_keys ++
+              [
+                store: %PacketStore{},
+                state: :initial_wait
+              ]
+
+  @doc """
+  Creates a new `t:#{inspect(__MODULE__)}.t/0`.
+  """
+  @spec new(options()) :: t()
+  def new(opts \\ []) do
+    %__MODULE__{
+      latency: opts[:latency] || @default_latency_ms
+    }
+  end
+
+  @doc """
+  Places a packet in the JitterBuffer.
+
+  Note: The initial latency timer will be set after the first packet is inserted into the buffer.
+  If you want to start it at your own discretion, schedule a `handle_timer/1` call prior to that.
+  """
+  @spec place_packet(t(), Packet.t()) :: result()
+  def place_packet(buffer, packet)
+
+  def place_packet(%{state: :initial_wait} = buffer, packet) do
+    {buffer, timer} = maybe_set_timer(buffer)
+    {_result, buffer} = try_insert_packet(buffer, packet)
+
+    {[], timer, buffer}
+  end
+
+  def place_packet(buffer, packet) do
+    case try_insert_packet(buffer, packet) do
+      {:ok, buffer} -> send_packets(buffer)
+      {:error, buffer} -> {[], nil, buffer}
+    end
+  end
+
+  @doc """
+  Flushes all remaining packets and resets the JitterBuffer.
+
+  Note: After flushing, the rollover counter is reset to `0`.
+  """
+  @spec flush(t()) :: result()
+  def flush(buffer) do
+    packets =
+      buffer.store
+      |> PacketStore.dump()
+      |> records_to_packets()
+
+    {packets, nil, %__MODULE__{latency: buffer.latency}}
+  end
+
+  @doc """
+  Handles the end of a previously set timer.
+  """
+  @spec handle_timer(t()) :: result()
+  def handle_timer(buffer) do
```

Comment:
Nitpick, I believe that in most places we use `handle_timeout`
```suggestion

  def handle_timeout(buffer) do
```